src/fmnoise/coldbrew.clj

```diff
@@ -1,38 +1,57 @@
 (ns fmnoise.coldbrew
-  (:import (com.github.benmanes.caffeine.cache CacheLoader LoadingCache Caffeine)
+  (:import (com.github.benmanes.caffeine.cache Cache CacheLoader LoadingCache Caffeine)
            (java.time Duration)))
 
 (defn cached
   "Accepts a function and creates cached version of it which uses Caffeine Loading Cache.
-  Cache expiration and refresh time (in seconds) can be provided as `:expire` and `:refresh` meta to function.
-  Example:
-  ```
+  Cache options should can be provided as meta to function. Supported options are:
+  `:expire` - expiration time (in seconds)
+  `:refresh` - refresh time (in seconds)
+  `:when` - function for checking if value should be cached
+
+  Examples:
   (cached ^{:expire 86400 :refresh 36000}
    (fn [db id] (query db id)))
-  ```"
+
+  ;; calculate todays order capacity until it reaches zero
+  (cached ^{:when zero?}
+   (fn [db worker-id] (order-capacity db worker-id)))
+  "
   [f]
-  (let [cache-builder (Caffeine/newBuilder)
+  (let [^Caffeine cache-builder (Caffeine/newBuilder)
+        condition-fn (-> f meta :when)
         _ (when-let [expire (-> f meta :expire)]
             (.expireAfterWrite cache-builder (Duration/ofSeconds expire)))
         _ (when-let [refresh (-> f meta :refresh)]
             (.refreshAfterWrite cache-builder (Duration/ofSeconds refresh)))
-        cache (.build cache-builder (reify CacheLoader (load [_ key] (apply f key))))]
-    (fn [& args] (.get ^LoadingCache cache args))))
+        cache (if condition-fn
+                (.build cache-builder)
+                (.build cache-builder (reify CacheLoader (load [_ key] (apply f key)))))]
```

Comment:
🤔 
If caffeine really doesn't cache `nil`, maybe this will work better?
```clj
(let [cache-loader (if condition-fn
                     (reify CacheLoader (load [_ key] (let [res (apply f key)] (when (condition-fn res) res))))
                     (reify CacheLoader (load [_ key] (apply f key))))]
(.build cache-builder cache-loader)
```
But of course, we should keep in mind will it change someday? (`nil`s caching)

Reply:
I'd be more explicit